Updated plotting themes and added functions

erika-tyagi opened the pull request:
Updated: 
* `theme_behindbars` to make it more modular and closer to the style guide 

Added: 
* `get_metric_description` 
* `is_fac_name` 
* `plot_recent_fac_increases` (took Neal's script, turned it into a function, then added a few extra parameters)  
* `plot_fac_trend` 

Function Questions: 
* `get_metric_description`: Mostly pulled from the website, but I'm not sure if all of these descriptions are accurate? (e.g. is `Residents.Tadmin` cumulative?) I'm also not sure if I should include other metrics, let me know! 
* `is_fac_name`: I thought about including a `verbose` parameter to output more info (e.g. what the match was, in which file, etc.), but this didn't seem like how most of the other functions were written, so decided against! Let me know if that would be helpful though! 
* `plot_recent_fac_increases`: Are there other parameters we would want to include? 
* `plot_recent_fac_trend`: Same thing, other parameters we would want to include? 

R Package Questions: 
* I wrote the R scripts, then ran `devtools::document()` to generate the documentation - is there anything else I have to do for the package build? Love to avoid breaking things if possible!!  
* For the package imports, I used `NCmisc::list.functions.in.file()` to figure out what to import - is there a better way? Also, I arbitrarily decided that if I was importing >=5  functions from a package, I'd just import the whole thing in that function (e.g. in `dplyr` and `ggplot2`). Is this okay?

hjohns12 commented on R/get_metric_description.R:
```diff
@@ -0,0 +1,54 @@
+        "Residents.Deaths" = "Cumulative COVID-19 Deaths", 
+        "Staff.Deaths" = "Cumulative COVID-19 Deaths", 
+        "Residents.Tadmin" = "COVID-19 Tests" 
+    )
```
Note to self: add description for ambiguous resident tests, present in historical data. This is called `Residents.Tested`

hjohns12 commented:
Overall style thing: I try to end each function with an explicit `return()` statement. This helps me look back and always know what the function's output / goal is. Totally up to you whether or not to implement, but I like it!

> Function Questions:
> 
> * `get_metric_description`: Mostly pulled from the website, but I'm not sure if all of these descriptions are accurate? (e.g. is `Residents.Tadmin` cumulative?) I'm also not sure if I should include other metrics, let me know!

In [this chart](https://twitter.com/uclaprisondata/status/1331638355184893953/photo/1) we used the label, "Cases among incarcerated people" for Residents.Confirmed. This almost seems like something in between the short and long description. It's probably overkill to add another description, but maybe we can tweak the long ones to mirror this format.

> * `is_fac_name`: I thought about including a `verbose` parameter to output more info (e.g. what the match was, in which file, etc.), but this didn't seem like how most of the other functions were written, so decided against! Let me know if that would be helpful though!

We added a `debug` parameter to the `clean_facility_name function` [here](https://github.com/uclalawcovid19behindbars/behindbarstools/blob/bcbab4a59ba21b3c9c76f2ac6256f0aab5a51724/R/clean_facility_name.R#L87) that does something similar. I think you could add a `debug` parameter (to follow the same conventions) here if you wanted to, although I couldn't really see myself using it. Maybe leave it for now, and we can add it in later if it becomes necessary?

> * `plot_recent_fac_increases`: Are there other parameters we would want to include?

Maybe something about how date breaks work? I've usually been adding in this to my plotting code: 
`scale_x_date(date_minor_breaks = "1 month", date_labels = "%m/%y", date_breaks = "1 month")`, but maybe we should make this optional.

> * `plot_recent_fac_trend`: Same thing, other parameters we would want to include?
Same as the above.

> 
> R Package Questions:
> 
> * I wrote the R scripts, then ran `devtools::document()` to generate the documentation - is there anything else I have to do for the package build? Love to avoid breaking things if possible!!

I think this is all you need, but I defer to @nmmarquez 

> * For the package imports, I used `NCmisc::list.functions.in.file()` to figure out what to import - is there a better way? Also, I arbitrarily decided that if I was importing >=5  functions from a package, I'd just import the whole thing in that function (e.g. in `dplyr` and `ggplot2`). Is this okay?

Sounds good to me!

nmmarquez commented on R/plot_recent_fac_increases.R:
```diff
@@ -0,0 +1,78 @@
+        ggplot(aes(x = Date, y = !!sym(metric), color = Title, label = last_value)) +
+        geom_line(size = 2.0) +
+        geom_point(size = 3.0, show.legend = F) +
+        {if (annotate) 
```
Did not know you could do if statements like this in the piping context thats so sick!

nmmarquez commented:
@erika-tyagi after you do `devtools::document` you should run `devtools::check` which will run the examples and check for errors in the package build. I pulled the branch and it looks like there are a few errors including needing to add `ggrepel` to the Imports list and the ambigous reference to scrape_df in the examples. Because some of the examples take a while to run and will continue to do so until `group_by_coalesce` is improved im going to suggest we put the [dontrun flag](https://stackoverflow.com/questions/12038160/how-to-not-run-an-example-using-roxygen2) on the examples which require a coalesce and the full data set so we dont have to wait 5+ minutes to check if anything broke. 

For the plotting functions I think the date formatting could go either way. If for whatever reason we do decide to not use the strategy Hope has suggested I think we should at least include an example of adding that to the returned plot in the documentation.

nmmarquez commented:
![2020-12-21-125203_566x119_scrot](https://user-images.githubusercontent.com/5198764/102806735-6b05f600-438b-11eb-9bee-ba91be336afb.png)

Just got through the checks after making some changes we could probably get away with just putting a dont run flag on the example in `plot_recent_fac_increases`

erika-tyagi commented:
Sweet thanks, very helpful feedback! I also learned that `devtools::check()` calls `devtools::document()`, so yay for one fewer step! Updates: 
* Added explicit return statements 
* Added `dontrun` to the examples that `group_by_coalesce` 
* Specified the date formatting (`"%b %d"` per the style guide) in both plots 
* Added examples to the docs showing how to layer onto plot function outputs since they're ggplot objects, e.g. 
```
plot_recent_fac_increases() + 
    scale_x_date(date_labels = "%m/%y") + 
    labs(y = "Cases Among Incarcerated People")
```
* Made the labs from `get_metric_description` a param called `auto_label` that's defaults to F  
* Fixed the stuff that `devtools::check()` yelled at me for 

Let me know of other thoughts, comments, issues, etc.

hjohns12 commented:
This looks great to me! I'd say it's ready to merge!

nmmarquez commented:
Agreed going to merge!
